### PR TITLE
Improvements to align CTS and Spec for Enqueue

### PR DESCRIPTION
**Pull request by martygrant**
- Add test for invalid null pointer for urEnqueueKernelLaunch and urEnqueueUSMFill
- Remove redundant sentence in urEnqueueMemBufferMap description
- Add advice flag parameters in test for urEnqueueUSMAdvise
- Added blocking/non-blocking success tests for urEnqueueDeviceGlobalVariableRead/Write
- Added all invalid size scenarios for urEnqueueMemBufferFill and urEnqueueMemBufferWrite/Read/CopyRect
- Added to description of urEnqueueMemAdvice to say some adapters will return UR_RESULT_ERROR_ADAPTER_SPECIFIC and allowing this return value in relevant tests

I've not attached an intel/llvm tag update as the spec change is very minor - just removing a sentence from an entry-point description.

### CODE REVIEW

File: test/conformance/enqueue/urEnqueueDeviceGlobalVariableRead.cpp [hunk 1]
```diff
@@ -29,10 +35,17 @@ TEST_P(urEnqueueDeviceGetGlobalVariableReadTest, Success) {
       queue, program, global_var.name.c_str(), true, sizeof(global_var.value),
       0, &global_var.value, 0, nullptr, nullptr));
 
+  if (!is_blocking) {
+    ASSERT_SUCCESS(urQueueFinish(queue));
+  }
+
   // kernel should increment value
```
Comment (isaacault) [inline test/conformance/enqueue/urEnqueueDeviceGlobalVariableRead.cpp:42]: I don't really understand what's happening in this test. Could you help by explaining

1. Shouldn't the `true` on line 35 be changed to `is_blocking`? 
2. What is the value of waiting on the queue to finish, and why only when non-blocking?

Comment (martygrant) [reply test/conformance/enqueue/urEnqueueDeviceGlobalVariableRead.cpp:42]: 1. yes it should, thanks!
2. it's asynchronous when non-blocking right? the waiting is just to make sure the read is completed before we assert its value, let me know if I've misunderstood

File: test/conformance/enqueue/urEnqueueDeviceGlobalVariableWrite.cpp [hunk 2]
```diff
@@ -5,6 +5,46 @@
...
+  // execute the kernel
+  ASSERT_SUCCESS(urEnqueueKernelLaunch(queue, kernel, n_dimensions,
+                                       &global_offset, &global_size, nullptr, 0,
+                                       nullptr, nullptr));
+  ASSERT_SUCCESS(urQueueFinish(queue));
+
+  // read global var back to host
+  int return_value = 0;
+  ASSERT_SUCCESS(urEnqueueDeviceGlobalVariableRead(
+      queue, program, global_var.name.c_str(), is_blocking, sizeof(return_value), 0,
```
Comment (isaacault) [inline test/conformance/enqueue/urEnqueueDeviceGlobalVariableWrite.cpp:41]: I think here `is_blocking` should be `true`.

File: test/conformance/enqueue/urEnqueueDeviceGlobalVariableRead.cpp (hunk 1 above)
Comment (isaacault) [reply test/conformance/enqueue/urEnqueueDeviceGlobalVariableRead.cpp]: That makes sense! Thanks

**Review by isaacault — APPROVED** [review]
